File: java/basics/FizzBuzz.java
// Print FizzBuzz for 1 through 100, with the label returned by a helper method.
// Run: java FizzBuzz.java

public class FizzBuzz {
    static String label(int n) {
        if (n % 15 == 0) {
            return "FizzBuzz";
        }
        if (n % 3 == 0) {
            return "Fizz";
        }
        if (n % 5 == 0) {
            return "Buzz";
        }
        return Integer.toString(n);
    }

    public static void main(String[] args) {
        for (int n = 1; n <= 100; n++) {
            System.out.println(label(n));
        }
    }
}

File: csharp/basics/FizzBuzz.cs
// Print FizzBuzz for 1 through 100, choosing the label with a switch expression on a tuple.
// Run: dotnet run FizzBuzz.cs (file-based app, .NET 10 or later), or as Program.cs in a console project

for (var n = 1; n <= 100; n++)
{
    Console.WriteLine((n % 3, n % 5) switch
    {
        (0, 0) => "FizzBuzz",
        (0, _) => "Fizz",
        (_, 0) => "Buzz",
        _ => n.ToString(),
    });
}
